### PR TITLE
Update some material button docs

### LINKED ISSUE #98691: Clarify ButtonThemeData usage and documentation
Feedback from @rydmike in https://github.com/flutter/flutter/issues/98537#issuecomment-1042524899

## ButtonThemeData

The `ButtonThemeData` class is marked as obsolete in the docs, but not as deprecated.

In its theme `ButtonTheme`:
https://github.com/flutter/flutter/blob/adafd66d9cc7796214404c8117395e207bf23d01/packages/flutter/lib/src/material/button_theme.dart#L53

And in its theme data `ButtonThemeData`:
https://github.com/flutter/flutter/blob/adafd66d9cc7796214404c8117395e207bf23d01/packages/flutter/lib/src/material/button_theme.dart#L170

In addition to the buttons that will be removed [FlatButton, OutlineButton, RaisedButton], the theme and theme data is used by the buttons' parent class `MaterialButton` and the `ButtonBar` and its theme `ButtonBarTheme`, plus as far I could notice `DropdownButton` uses one property from it as well.

Instead of having the docs above mention that it is used by buttons that will no longer exist, maybe consider updating the doc comment and instead mention the usage by `ButtonBar`, `DropdownButton` and `MaterialButton`, if it will remain. It kind of will have to, but more on that below.

Perhaps for the update also consider updating the doc comment for the `buttonTheme` property in `ThemeData` to better reflects its remaining usage in `ButtonBar`, `DropdownButton` and `MaterialButton`

**Some of these doc updates will be completed in the course of resolving #98537**

https://github.com/flutter/flutter/blob/adafd66d9cc7796214404c8117395e207bf23d01/packages/flutter/lib/src/material/theme_data.dart#L1322

---

In general, I think the current use case is unclear to users, `obsolete` is not deprecated, but it's not really `obsolete` either if other parts of the framework still use ButtonThemeData. There is definitely some opportunity for better clarity and guidance here - even if that is just to say we plan to make more changes in the future that will eventually remove this class.

cc @HansMuller @darrenaustin

### LINKED ISSUE #98692: Fix ButtonStyleButton docs
From @rydmike in https://github.com/flutter/flutter/issues/98537#issuecomment-1042524899

In `ButtonStyleButton` there is also a doc ref to `MaterialButton` that I suspect should be to `ButtonStyleButton` instead, or just as a normal text "Material button" as a reference to the Material button concept itself, rather than a doc ref to a class that has nothing to do with `ButtonStyleButton` where the reference is.

https://github.com/flutter/flutter/blob/8a6d0d62f3257e74e2850e68bd97b98877c78e2e/packages/flutter/lib/src/material/button_style_button.dart#L425

### LINKED ISSUE #98693: Clarify MaterialButton usage and documentation
From @rydmike in https://github.com/flutter/flutter/issues/98537#issuecomment-1042524899

## MaterialButton

Next comes the question of the fate of the `MaterialButton` the parent class for the buttons to be removed.

It is also in the docs marked as **obsolete** but not deprecated. 

https://github.com/flutter/flutter/blob/adafd66d9cc7796214404c8117395e207bf23d01/packages/flutter/lib/src/material/material_button.dart#L22

Will it remain available or be removed? My conclusion is, it will stay, even if it does not correctly implement or enable users to create Material buttons, plus its children will be gone due to same reasons. Considering this, it seems a bit illogical for it to remain, but it would be difficult to remove without a number of other changes discussed further below.

My search did no reveal any other usages and mentions of `MaterialButton` than in its child buttons that will be removed, and in `ButtonThemeData` itself, in a large number of odd looking getters.

<br>

> Removed quoted section here as a separate issue: https://github.com/flutter/flutter/issues/98692

<br>

The fact that `ButtonThemeData`, that needs to remain for `ButtonBar` and `DropdownButton`, itself depends on `MaterialButton`, also makes it impossible to remove the "obsolete" `MaterialButton` parent of all the 3 buttons now being removed.

In `Material` there is in the `MaterialType` enum a doc comment to `MaterialButton` in enum value `button`. 
https://github.com/flutter/flutter/blob/8a6d0d62f3257e74e2850e68bd97b98877c78e2e/packages/flutter/lib/src/material/material.dart#L35

So I checked for references to `MaterialType.button`, other than in `Material` it is only used by `RawMatererialButton` and `ButtonStyleButton`. maybe consider updating the above doc ref to them instead of `MaterialButton`.

If the `MaterialButton` is considered and marked as **obsolete** in the docs, which imo is correct, thus maybe consider deprecating it and plan for changes need to be able to remove it.

---

Similar to #98691, `obsolete` is not a term folks know what to do with. What does it mean when other parts of the framework still depend on it? Some guidance would be helpful to add. cc @HansMuller & @darrenaustin

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/98692

Maybe fixes https://github.com/flutter/flutter/issues/98693

If this is a good improvement to MaterialButton, I can do the same to resolve https://github.com/flutter/flutter/issues/98691

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/material_button.dart [hunk 1]
```diff
@@ -462,4 +453,5 @@ class MaterialButton extends StatelessWidget {
 ///
 /// This mixin only exists to give the "label and icon" button widgets a distinct
 /// type for the sake of [ButtonTheme].
+// TODO(Piinks): Should this be deprecated?
```
Comment (Piinks) [inline packages/flutter/lib/src/material/material_button.dart:456]: This is only used by the buttons that are being removed, should it be deprecated @HansMuller? I checked customer references as well, no one is using this.

**Comment by Piinks** [comment]
@rydmike I would love to know what you think of this :)

File: packages/flutter/lib/src/material/material_button.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/material_button.dart:456]: Yes

**Comment by Piinks** [comment]
Caught up with @HansMuller offline, the customizations afforded by MaterialButton and RawMaterialButton are achievable in the new button universe, and so we should deprecate the classes that are currently described as obsolete. This will mean we need to migrate and update more aspects of the framework, but to provide clarity for users we are going to start this deprecation period to direct folks to the right API to prevent further confusion.

**Comment by Piinks** [comment]
I am going to close this in favor of a new PR

**Pull request closed without merging**